### ISSUE TITLE
RKE2 cluster provisioning broken since v1.21.8 release

### ISSUE THREAD

**Issue by janeczku** [rancher/rancher#36051]
**Rancher Server Setup**
- Rancher version: 2.6.3 HA

**Information about the Cluster**
- Kubernetes version: v1.21.8+rke2r1  / v1.22.5+rke2r1

**Describe the bug**

Provisioning of multi-node RKE2 clusters never succeeds. All but the first node get stuck in status "waiting on probes".

This issue is a regression from the previous release `v1.21.7+rke2r2` which works fine.

<img width="750" alt="Screenshot 2022-01-06 at 13 26 49" src="https://user-images.githubusercontent.com/3813921/148385904-3bc1f198-efd2-4437-9e81-4bfd69002f83.png">

**To Reproduce**

- Provision RKE2 cluster with an infrastructure driver (reproduced on Digital Ocean, Harvester)
- Select the latest non-experimental version, `v1.21.8+rke2r1`
 - Configure a machine pool with 3 nodes using all roles (cp, etcd, worker)

**Additional Information**

rke2-server log on first node:

```
level=error msg="Failed to authenticate request from 10.65.0.108:40342: invalid username/password combination"
```

rke2-server log on additional nodes:

```
level=fatal msg="starting kubernetes: preparing server: https://10.65.0.112:9345/v1-rke2/config: 401 Unauthorized"
```

**Comment by jsompis** [comment]
I also have this issue.

**Comment by amincheloh** [comment]
confirm this issue on custom nodes

**Comment by TachunLin** [comment]
This issue can be reproduced on provo bare machines with `v1.21.8+rke2r1` on rancher 2.6.3

![image](https://user-images.githubusercontent.com/29251855/148524221-1fb8d376-fb3a-4991-8e5f-58db6d55e2a7.png)

![image](https://user-images.githubusercontent.com/29251855/148525072-13301382-f867-4a91-a563-e3142d2b5b9a.png)

On the running node:
```
ubuntu@rke2-cluster-1-21-8-pool1-d00bb5f8-5cs72:~$ journalctl -u rke2-server -f
-- Logs begin at Fri 2022-01-07 08:53:42 UTC. --
Jan 07 09:30:18 rke2-cluster-1-21-8-pool1-d00bb5f8-5cs72 rke2[2358]: time="2022-01-07T09:30:18Z" level=error msg="Failed to authenticate request from 10.84.45.149:52250: invalid username/password combination"
```
### Additional context

* The kubeconfig file 
[kubeconfig.zip](https://github.com/rancher/rancher/files/7827960/kubeconfig.zip)

* The cluster yaml file
[cluster.zip](https://github.com/rancher/rancher/files/7827961/cluster.zip)

### Test Information
* Test Environment: 3 nodes harvester on provo bare-metal machine
* Harvester version: v1.0.0
* Rancher version: v2.6.3

**Comment by ekarlso** [comment]
Is there any workaround for this?

**Comment by amincheloh** [comment]
@ekarlso create a cluster then edit as YAML and change kubernetesVersion to v1.21.7+rke2r2. After all nodes ready, upgrade to v1.21.8+rke2r1

**Comment by moosicplexix** [comment]
Also seeing this. @amincheloh I did what you suggested and my first node is reporting this when I tried to join it to the cluster:

`401 received while downloading Rancher connection information. Sleeping for 5 seconds and trying again`

**Comment by bashofmann** [comment]
This issue is fixed with `v1.21.8+rke2r2`.

**Comment by timhaneunsoo** [comment]
### Test Environment: ###
Rancher version: v2.6-head 19f459a
Rancher cluster type: HA
Docker version: 20.10

Downstream cluster type: Digital Ocean

----------

### Testing: ###

Tested this issue with the following steps:
1. Provision RKE2 cluster with an infrastructure driver (Digital Ocean)
2. Select the latest non-experimental version, v1.21.8+rke2r1
  - Configure a machine pool with 3 nodes using all roles (cp, etcd, worker)

**Result** - Pass

Single pool scenario used in the bug description is provisioned successfully for both v1.21 and v1.22
**v1.21**
![image.png](https://zube.io/files/rancher/14069fb65416ee4112ad477b3e5142bc-image.png)
**v1.22**
![image.png](https://zube.io/files/rancher/ec2abcac4b0b04516593b45599d529dc-image.png)

Multi pool scenario is provisioned successfully
**v1.21**
![image.png](https://zube.io/files/rancher/28e5a4432dec1274a05960a1ec833a50-image.png)
**v1.22**
![image.png](https://zube.io/files/rancher/a460dffb407dc32c3fd049dfe09acdee-image.png)

Fresh Install - Pass
Upgrade Install v1.21.6 -> v1.21.8 - Pass
